src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala

```diff
@@ -241,20 +248,20 @@ object DockerPlugin extends AutoPlugin {
         }
       },
       sourceDirectory := sourceDirectory.value / "docker",
-      stage := Stager.stage(Docker.name)(streams.value, stagingDirectory.value, mappings.value),
+      stage := Stager.stage(Docker.name)(streams.value, stagingDirectory.value, dockerLayerMappings.value.map {
+        case (_, file, path) => (file, path)
+      }),
       stage := (stage dependsOn dockerGenerateConfig).value,
       stagingDirectory := (target in Docker).value / "stage",
       dockerLayerMappings := {
         val dockerGroups = dockerLayerGrouping.value
-        def buildDockerPaths(seq: Seq[(File, String)], prefix: String) =
-          for {
-            (file, path) <- seq
-            layerIdx = dockerGroups(path)
-            pathInDocker = s"$prefix/$layerIdx/$path"
-          } yield (layerIdx, file, pathInDocker)
-
-        buildDockerPaths((mappings in Universal).value, (defaultLinuxInstallLocation in Docker).value + "/") ++
-          buildDockerPaths(dockerPackageMappings.value, "")
+        val dockerFinalFiles = (mappings in Docker).value
+        for {
+          (file, path) <- dockerFinalFiles
+          layerIdx = dockerGroups(path)
+          dockerLayerPath = s"/$layerIdx$path"
+        } yield (layerIdx, file, dockerLayerPath)
```

Comment:
It would probably be better to use `(layerIdx, file, path)` instead, since we can always build `dockerLayerPath` from that where we need it. That way we won't have to replace `.startsWith` with less preciese `.contains` in `dockerAdditionalPermissions `.